src/conf_mode/system_frr.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2021 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from pathlib import Path
+from re import compile as re_compile
+from re import M as re_M
+from sys import exit
+
+from vyos import ConfigError
+from vyos import airbag
+from vyos.config import Config
+from vyos.logger import syslog
+from vyos.util import read_file, write_file, run
+airbag.enable()
+
+# path to daemons config and config status files
+config_file = '/etc/frr/daemons'
+vyos_status_file = '/tmp/vyos-config-status'
+# path to watchfrr for FRR control
+watchfrr = '/usr/lib/frr/watchfrr.sh'
+
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+    base = ['system', 'frr']
+    if not conf.exists(base):
+        return {}
+
+    frr_config = conf.get_config_dict(base)
+
+    return frr_config
+
+
+def daemons_config_parse(daemons_config):
+    # create regex for parsing daemons options
+    regex_daemon_config = re_compile(
+        r'^(?P<daemon_name>\w+)_options="(?P<daemon_options>.*)"$', re_M)
+    # create empty dict for config
+    daemons_config_dict = {}
+    # fill dictionary with actual config
+    for daemon in regex_daemon_config.finditer(daemons_config):
+        daemon_name = daemon.group('daemon_name')
+        daemon_options = daemon.group('daemon_options')
+        daemons_config_dict[daemon_name] = daemon_options
+
+    # return daemons config
+    return (daemons_config_dict)
+
+
+def verify(frr_config):
+    # Nothing to verify here
+    pass
+
+
+def generate(frr_config):
+    # read daemons config file
+    daemons_config = read_file(config_file)
```

Comment:
Please do not read/modify/write the daemon file. We always render a full template which gives less possibilities of errors. Also that is the "standard" way.

Reply:
Sure, this makes the task even easier. Done.